### PR TITLE
TS migration

### LINKED ISSUE #1: Migrate to TS and update README
@jobo322 

Could you:
- add types
- change README to have correct examples
- have similar API to https://github.com/mljs/regression-polynomial/blob/main/src/index.ts

This can then be used for your 2D baseline correction.

### PR DESCRIPTION
**Pull request by jobo322**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,14 +1,19 @@
 {
   "compilerOptions": {
-    "allowJs": true,
     "esModuleInterop": true,
     "moduleResolution": "node",
-    "outDir": "lib",
-    "sourceMap": true,
+    "declaration": true,
```
Comment (lpatiny) [inline tsconfig.json]: @targos I would like your feedback on the changes in tsconfig.json

File: tsconfig.json [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
+    "declarationDir": "types",
```
Comment (targos) [inline tsconfig.json]: we don't use this normally

File: tsconfig.json [hunk 3]
```diff
@@ -1,14 +1,19 @@
 {
   "compilerOptions": {
-    "allowJs": true,
     "esModuleInterop": true,
     "moduleResolution": "node",
-    "outDir": "lib",
```
Comment (targos) [inline tsconfig.json:6]: where does the build go without an outDir?

File: tsconfig.json [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
-    "sourceMap": true,
+    "declaration": true,
+    "declarationDir": "types",
     "strict": true,
-    "target": "es2020"
+    "target": "es2020",
```
Comment (targos) [inline tsconfig.json]: es2022

File: tsconfig.json [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
+    "skipLibCheck": true,
```
Comment (targos) [inline tsconfig.json:7]: we should only enable this if there's an issue with a dependency we don't control

File: tsconfig.json [hunk 6]
```diff
@@ -1,14 +1,19 @@
...
+    "resolveJsonModule": true,
```
Comment (targos) [inline tsconfig.json]: are there json files imported in the project?

File: src/PolynomialRegression2D.ts [hunk 7]
```diff
@@ -1,165 +1,139 @@
...
+  constructor(
+    inputs: DataXY,
+    outputs: NumberArray,
+    options: PolynomialRegression2DOptions = {},
+  ) {
     super();
+    // @ts-expect-error internal use only
     if (inputs === true) {
-      // reloading model
+      // @ts-expect-error internal use only
```
Comment (lpatiny) [inline src/PolynomialRegression2D.ts:42]: @targos In order to reload the regression result we use this method that does not seem that nice to me (need to add ts-expect-error).

Could `static load` rather then create a `new PolynomialRegression2D` change the prototype of the object or another way ?

Comment (targos) [reply src/PolynomialRegression2D.ts:42]: We do it like that in other packages. I don't think it's a good idea to make hacks around prototypes.

File: tsconfig.json (hunk 5 above)
Comment (jobo322) [reply tsconfig.json:7]: not sure if it is the source of the problem but NodeJS throws an error like:

Cannot find namespace 'NodeJS'

File: tsconfig.json [hunk 8]
```diff
@@ -3,11 +3,8 @@
     "esModuleInterop": true,
     "moduleResolution": "node",
     "declaration": true,
-    "declarationDir": "types",
-    "strict": true,
```
Comment (targos) [inline tsconfig.json:8]: strict is useless???

**Pull request merged**